azure-pipelines.yml

```diff
@@ -53,22 +53,22 @@ stages:
             displayName: Terraform validate
             inputs:
               command: validate
-              workingDirectory: $(System.DefaultWorkingDirectory)/terraform
+              workingDirectory: $(System.DefaultWorkingDirectory)/terraform/${{ environment }}
 
           - task: TerraformCLI@0
             displayName: Terraform plan
             inputs:
               command: plan
-              workingDirectory: $(System.DefaultWorkingDirectory)/terraform
+              workingDirectory: $(System.DefaultWorkingDirectory)/terraform/${{ environment }}
```

Comment:
I think it would be better if we didn't add per environment folders for terraform.

With this setup it means:
1. You can't easily add the config from web-ui only because it's two files
2. We may get issues with teams only adding non prod config and coming back for prod later or just forgetting it, easier if we make sure they do all envs as part of onboarding.

Instead we could run it twice with a different environment parameter.
To keep it contained to one file we could do:

```terraform
locals {
 rd_non_prod_domains = ["mail-rd-nonprod.platform.hmcts.net"]
 rd_prod_domains = ["mail-rd.platform.hmcts.net"]
}

module "reference-data" {
  source      = "../modules"
  environment = var.environment
  account     = "reference-data"
  domains     = var.environment == "prod" ? local.rd_prod_domains : local.rd_non_prod_domains
}
````

Reply:
Sure - make sense, I have now changed it to the structure above..